### PR TITLE
Make typing more complete

### PR DESCRIPTION
**Pull request by KOLANICH**

### CODE REVIEW

File: probables/blooms/bloom.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 from ..exceptions import InitializationError, NotSupportedError
 from ..hashes import HashFuncT, HashResultsT
 from ..utilities import is_hex_string, is_valid_file
-from .basebloom import BaseBloom
+from .basebloom import BaseBloom, HashFuncT
```
Comment (barrust) [inline probables/blooms/bloom.py:16]: Not sure why `HashFuncT` is imported here as it is pulled in from the `hashes` utility file.

File: probables/blooms/countingbloom.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
 MISMATCH_MSG = "The parameter second must be of type CountingBloomFilter"
 
 
-def _verify_not_type_mismatch(second):
+def _verify_not_type_mismatch(second: typing.Union[int, "CountingBloomFilter"]) -> bool:
```
Comment (barrust) [inline probables/blooms/countingbloom.py:16]: This  does not need to be a union of int and "CountingBloomFilter". Just "CountingBloomFilter" should be sufficient.

File: probables/blooms/expandingbloom.py [hunk 3]
```diff
@@ -11,7 +11,7 @@
 from ..exceptions import RotatingBloomFilterError
 from ..hashes import HashFuncT, HashResultsT, KeyT
 from ..utilities import is_valid_file
-from .bloom import BloomFilter
+from .bloom import BloomFilter, HashFuncT
```
Comment (barrust) [inline probables/blooms/expandingbloom.py:14]: Not sure why `HashFuncT` is imported here as it is pulled in from the `hashes` utility file.

File: probables/exceptions.py [hunk 4]
```diff
@@ -43,7 +43,7 @@ class CuckooFilterFullError(ProbablesBaseException):
     Args:
         message (str): The error message to be reported"""
 
-    def __init__(self, message: str):
+    def __init__(self, message: str) -> str:
```
Comment (barrust) [inline probables/exceptions.py:46]: I believe this should be `-> None` and not `-> str`

**Review by barrust — CHANGES_REQUESTED** [review]
Overall this is great, thanks! Do you think it is necessary to type hint private functions?

File: probables/blooms/countingbloom.py (hunk 2 above)
Comment (KOLANICH) [reply probables/blooms/countingbloom.py:16]: This reflects the actual usage. The first PR about typing has been created semi-automatically by using `monkeytype` tool that records what every function actually returns and then examining the changes and refactoring them.

This PR is just rebase of that one with with conflicts solved and some additional refactoring.

File: probables/exceptions.py (hunk 4 above)
Comment (KOLANICH) [reply probables/exceptions.py:46]: You are right,

**Comment by KOLANICH** [comment]
>Do you think it is necessary to type hint private functions?

I think the code should be hjnted as much as possible bevause it makes the code more readable and improves autocompletion using language servers.

File: probables/blooms/countingbloom.py (hunk 2 above)
Comment (barrust) [reply probables/blooms/countingbloom.py:16]: I think that is from a test to check that it correctly checks for the type and isn't truly allowing int as an input.

File: probables/blooms/expandingbloom.py (hunk 3 above)
Comment (KOLANICH) [reply probables/blooms/expandingbloom.py]: Fixed.

File: probables/blooms/bloom.py (hunk 1 above)
Comment (KOLANICH) [reply probables/blooms/bloom.py]: Fixed.

File: probables/blooms/countingbloom.py (hunk 2 above)
Comment (KOLANICH) [reply probables/blooms/countingbloom.py]: Fixed.

**Review by barrust — APPROVED** [review]

**Pull request merged**